src/Protocol/Protocol_1_14.cpp

```diff
@@ -104,8 +206,84 @@ void cProtocol_1_14::SendPaintingSpawn(const cPainting & a_Painting)
 
 
 
+void cProtocol_1_14::SendParticleEffect(const AString & a_ParticleName, float a_SrcX, float a_SrcY, float a_SrcZ, float a_OffsetX, float a_OffsetY, float a_OffsetZ, float a_ParticleData, int a_ParticleAmount)
+{
+	ASSERT(m_State == 3);  // In game mode?
+
+	cPacketizer Pkt(*this, pktParticleEffect);
+	Pkt.WriteBEInt32(GetProtocolParticleID(a_ParticleName));
+	Pkt.WriteBool(false);
+	Pkt.WriteBEFloat(a_SrcX);
+	Pkt.WriteBEFloat(a_SrcY);
+	Pkt.WriteBEFloat(a_SrcZ);
+	Pkt.WriteBEFloat(a_OffsetX);
+	Pkt.WriteBEFloat(a_OffsetY);
+	Pkt.WriteBEFloat(a_OffsetZ);
+	Pkt.WriteBEFloat(a_ParticleData);
+	Pkt.WriteBEInt32(a_ParticleAmount);
+}
+
+
+
+
+
+void cProtocol_1_14::SendParticleEffect(const AString & a_ParticleName, Vector3f a_Src, Vector3f a_Offset, float a_ParticleData, int a_ParticleAmount, std::array<int, 2> a_Data)
+{
+	ASSERT(m_State == 3);  // In game mode?
+
+	const auto ParticleID = GetProtocolParticleID(a_ParticleName);
+
+	cPacketizer Pkt(*this, pktParticleEffect);
+	Pkt.WriteBEInt32(ParticleID);
+	Pkt.WriteBool(false);
+	Pkt.WriteBEFloat(a_Src.x);
+	Pkt.WriteBEFloat(a_Src.y);
+	Pkt.WriteBEFloat(a_Src.z);
+	Pkt.WriteBEFloat(a_Offset.x);
+	Pkt.WriteBEFloat(a_Offset.y);
+	Pkt.WriteBEFloat(a_Offset.z);
+	Pkt.WriteBEFloat(a_ParticleData);
+	Pkt.WriteBEInt32(a_ParticleAmount);
+
+	switch (ParticleID)
+	{
+		// blockdust
+		case 3:
+		{
+			Pkt.WriteVarInt32(static_cast<UInt32>(a_Data[0]));
+			break;
+		}
+	}
+}
+
+
+
+
+
+void cProtocol_1_14::SendRespawn(eDimension a_Dimension)
+{
+	cPacketizer Pkt(*this, pktRespawn);
+	cPlayer * Player = m_Client->GetPlayer();
+	Pkt.WriteBEInt32(static_cast<Int32>(a_Dimension));
+	Pkt.WriteBEUInt8(static_cast<Byte>(Player->GetEffectiveGameMode()));
+	Pkt.WriteString("default");
+}
+
+
+
+
+
 void cProtocol_1_14::SendSoundParticleEffect(const EffectID a_EffectID, int a_SrcX, int a_SrcY, int a_SrcZ, int a_Data)
 {
+	ASSERT(m_State == 3);  // In game mode?
+
+	// Note: Particles from block break missing
+
+	cPacketizer Pkt(*this, pktSoundParticleEffect);
+	Pkt.WriteBEInt32(static_cast<int>(a_EffectID));
+	Pkt.WriteXYZPosition64(a_SrcX, a_SrcY, a_SrcZ);
+	Pkt.WriteBEInt32(a_Data);
+	Pkt.WriteBool(false);
```

Comment:
This is exactly the same code as some code it overrides. Was the method of writing the position changed, maybe? (XYZ ->  XZY)

Reply:
Same thing apply here for the call of GetProtocolParticleID.